editor/store/reducer.js

```diff
@@ -292,6 +292,36 @@ export const editor = flow( [
 				};
 			}
 
+			case 'MOVE_BLOCK_TO_INDEX': {
+				const { rootUID = '' } = action;
+				const subState = state[ rootUID ];
+
+				if ( ! Number.isInteger( action.index ) || ! subState.length ) {
+					return state;
+				}
+
+				const blockIndex = subState.indexOf( action.uid );
+
+				if ( blockIndex === -1 || blockIndex === action.index ) {
+					return state;
+				}
+
+				if ( action.index < 0 ) {
+					action.index = 0;
+				} else if ( action.index >= subState.length ) {
+					action.index = subState.length - 1;
+				}
+
+				const _state = [ ...subState ];
```

Comment:
What we're assigning here is not what we expect to become the state object (which includes all other root UIDs) and the `_` prefixing does not convey any meaning. I'd suggest `nextSubState` instead.

Reply:
Ok. I believe I was just following the changes made to the other reducers after adding nested blocks. I will keep in mind while refactoring this part (as it will be necessary to have it work with nested blocks).